### ISSUE TITLE
TODO for v1

### ISSUE THREAD

**Issue by ilaicraftYT** [OpenWars/resources#1]
We must create those resources before v1 release of OW

- [x] Main Menu
- [x] Terrain
- [ ] Units
- [ ] Buildings
- [ ] Backgrounds
- [ ] Animation Units
- [ ] Animation Backgrounds
- [ ] GUI Textures
- [ ] Music
- [ ] Sound Effects
- [ ] Battle HUD
- [ ] COs Full Body Characters
- [ ] COs Battle Expressions

**Comment by ilaicraftYT** [comment]
Music should be obtained by the player.

**Comment by ilaicraftYT** [comment]
Animation for v2

**Comment by rivest-oss** [comment]
> Music should be obtained by the player.

I managed to extract some songs of the game, and export it as MIDI, with a FOSS (GNU-licensed?) tool.
I think that tool is good enough for that

**Comment by ilaicraftYT** [comment]
> > Music should be obtained by the player.
> 
> I managed to extract some songs of the game, and export it as MIDI, with a FOSS (GNU-licensed?) tool. I think that tool is good enough for that

I think it's too much effort for a non-technical player to do it.

**Comment by rivest-oss** [comment]
> > > Music should be obtained by the player.
> > 
> > 
> > I managed to extract some songs of the game, and export it as MIDI, with a FOSS (GNU-licensed?) tool. I think that tool is good enough for that
> 
> I think it's too much effort for a non-technical player to do it.

Hm... would it better if I make an extractor for OW that extracts the music from the ROM? Like, "Select your ROM" (OpenRA-like).

**Comment by ilaicraftYT** [comment]
> > > > Music should be obtained by the player.
> > > 
> > > 
> > > I managed to extract some songs of the game, and export it as MIDI, with a FOSS (GNU-licensed?) tool. I think that tool is good enough for that
> > 
> > 
> > I think it's too much effort for a non-technical player to do it.
> 
> Hm... would it better if I make an extractor for OW that extracts the music from the ROM? Like, "Select your ROM" (OpenRA-like).

That would be a lot better, but we're going to include a whole MIDI extractor and player just for music?

Personally, I think it would be a lot better if we just name it "custom music" and let players play whatever they want, be it AW or not.

**Comment by rivest-oss** [comment]
> > > > > Music should be obtained by the player.
> > > > 
> > > > 
> > > > I managed to extract some songs of the game, and export it as MIDI, with a FOSS (GNU-licensed?) tool. I think that tool is good enough for that
> > > 
> > > 
> > > I think it's too much effort for a non-technical player to do it.
> > 
> > 
> > Hm... would it better if I make an extractor for OW that extracts the music from the ROM? Like, "Select your ROM" (OpenRA-like).
> 
> That would be a lot better, but we're going to include a whole MIDI extractor and player just for music?
> 
> Personally, I think it would be a lot better if we just name it "custom music" and let players play whatever they want, be it AW or not.

Well, that's true.
Maybe I'll make a stand-alone extractor/converter.
Anyway, it would be nice to add a MIDI player, and I think raylib actually implements something like that.

**Comment by rivest-oss** [comment]
P.s.:  No, it doesn't
> raylib already supports XM and MOD files. No plans to add MIDI support for the moment but contributions are welcome.
( https://old.reddit.com/r/raylib/comments/axus5b/midi_file_support/ )
> @zaher I considered supporting MIDI files, the actual issue with MIDI is the requirement for a sounds bank to play them, so, it would require adding one to the library just for MIDI... or at least adding specific API functions to configure one external... at that point, it's probably easier to just let the user manage MIDI loading and playing.
 ( https://github.com/raysan5/raylib/discussions/1502#discussioncomment-671232 )

**Comment by ilaicraftYT** [comment]
Yeah, a standalone extractor would be a lot better, but I don't think it's
worth the effort. Latest version of gba-mus barely gets to produce okay
MIDIs for me.
